src/v2/guide/accessibility.md

```diff
@@ -0,0 +1,264 @@
+  data: {
+    checked: false,
+  },
+})
+</script>
+{% endraw %}
+
+The `sr-only` class means that the checkbox is visible only to screen readers, and `aria-hidden="true"` on the toggle button means that it is hidden to screen readers. Screen readers read out the checkbox, and your sighted users see the toggle button as intended—it works for both parties.
+
+#### ARIA attributes
+
+We saw an ARIA attribute in the previous example to hide the custom HTML from the screen reader. aria-hidden is one of a tonne of attributes that you can add to elements to tell assistive technology such as screen readers what things on the page are.
+
+Check out MDN for a longer read on ARIA: [An overview of accessible web applications and widgets][aria mdn].
+
+To make first example accessible using ARIA, we have to set the role to "checkbox" and use the `aria-checked` attribute:
+
+```html
+<div
+  role="checkbox"
+  tabindex="0"
+  :aria-checked="checked"
+  :class="['toggle-button', { 'toggle-button--checked': checked }]"
+  @click="checked = !checked">
+  <div class="toggle-button__handle"></div>
+</div>
+```
+
+{% raw %}
+<div id="example-3" class="demo">
+  <div
+    role="checkbox"
+    tabindex="0"
+    :aria-checked="checked"
+    :class="['toggle-button', { 'toggle-button--checked': checked }]"
+    @click="checked = !checked">
+    <div class="toggle-button__handle"></div>
+  </div>
+  
+  <pre>{ checked: {{ checked }} }</pre>
+</div>
+<script>
+var example1 = new Vue({
+  el: '#example-3',
+  data: {
+    checked: false,
+  },
+})
+</script>
+{% endraw %}
+
+Either method of making the custom input element accessible works fine and isn't too difficult to implement.
+
+### Dynamic content
+
+when content is updated, screen readers don't know it has been updated
+
+examples:
+
+- error messages on forms
+- new information (eg news, notifications) from server
+- requested information finished downloading
+
+you can use aria-live to let the user know that stuff has changed
+
+
+### Page navigation with client-side routing
+
+when you click a link in a website with server-side routing, the browser knows the page has changed. on a website with client-side routing, the browser doesn't know, so it isn't announced to the user - it just announces that the link has been clicked, not that navigation has occured or what is displayed has changed.
+
+the solution is to announce to the user when the route has changed using a hidden aria-live element.
+
+## Testing accessibility
+
+- automated tools
+- screen reader - http://12devsofxmas.co.uk/2016/01/day-8-testing-using-a-screen-reader/ https://developer.paciellogroup.com/blog/2015/01/basic-screen-reader-commands-for-accessibility-testing/
+- disabled user testers
+
+
+[wcag 2.0]: https://www.w3.org/TR/WCAG20/
+[switch technology]: https://en.wikipedia.org/wiki/Switch_access
+[aria mdn]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/An_overview_of_accessible_web_applications_and_widgets
```

Comment:
I think axe-core is worth mentioning here

Reply:
I also added [WAVE](https://wave.webaim.org/), as it's one I use frequently (especially the browser extension).